src/utils/az.ts

```diff
@@ -254,26 +306,278 @@ export class Az implements AzApi {
+         };
+      }
+   }
+
+   async createRoleAssignment(
+      subscriptionItem: SubscriptionItem,
+      role: {name: string; id: string},
+      assignee: string,
+      scope: string
+   ): Promise<Errorable<RoleAssignmentItem>> {
+      const subscriptionId = subscriptionItem.subscription.subscriptionId;
+      if (subscriptionId === undefined) {
+         return {succeeded: false, error: 'SubscriptionId undefined'};
+      }
+
+      try {
+         const creds = this.getCreds();
+         const client = new AuthorizationManagementClient(
+            creds,
+            subscriptionId
+         );
+         const resp = await client.roleAssignments.create(scope, role.name, {
+            principalId: assignee,
+            roleDefinitionId: role.id
+         });
+         return {succeeded: true, result: {roleAssignment: resp}};
+      } catch (error) {
+         return {
+            succeeded: false,
+            error: `Failed to create role assignment: ${error}`
+         };
+      }
+   }
+
+   async getResource(
+      subscriptionItem: SubscriptionItem,
+      resourceId: string
+   ): Promise<Errorable<ResourceItem>> {
+      const subscriptionId = subscriptionItem.subscription.subscriptionId;
+      if (subscriptionId === undefined) {
+         return {succeeded: false, error: 'SubscriptionId undefined'};
+      }
+
+      try {
+         const creds = this.getCreds();
+         const client = new ResourceManagementClient(creds, subscriptionId);
+         const resource = await client.resources.getById(
+            resourceId,
+            LATEST_ARM_RESOURCE_VERSION
+         );
+         return {succeeded: true, result: {resource}};
+      } catch (error) {
+         return {
+            succeeded: false,
+            error: `Failed to get resource "${resourceId}": ${error}`
+         };
+      }
+   }
+
    private getContainerRegistryClient(
       creds: TokenCredential,
       loginServer: string,
-      environment: Environment
+      audience: KnownContainerRegistryAudience
    ): ContainerRegistryClient {
-      // @azure/container-registry doesn't support ADAL tokens at all and will error without this
-      // https://github.com/Azure/azure-sdk-for-js/issues/21192
-      (creds as any).signRequest = undefined;
-
       return new ContainerRegistryClient(`https://${loginServer}`, creds, {
-         audience: environment.resourceManagerEndpointUrl
+         audience
       });
    }
 }
 
-export function getAzureAccount(): AzureAccountExtensionApi {
-   return (<AzureExtensionApiProvider>(
-      vscode.extensions.getExtension('ms-vscode.azure-account')!.exports
-   )).getApi('1.0.0');
-}
+// Ideally we'd use the Azure Account Extension to authenticate but there's major bugs with it right now
+// https://github.com/Azure/azure-sdk-for-js/issues/22904
+export const getAzCreds: CredGetter = () => new DefaultAzureCredential();
```

Comment:
more of a question than feedback - where does the DefaultAzureCredential get its creds/login info from if the users aren't signing in themselves via the vscode extension?